homeassistant/components/indevolt/__init__.py

```diff
@@ -0,0 +1,32 @@
+"""Home Assistant integration for INDEVOLT devices."""
+
+from __future__ import annotations
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+
+from .const import PLATFORMS
+from .coordinator import IndevoltDeviceUpdateCoordinator
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Indevolt from a config entry."""
+    coordinator = IndevoltDeviceUpdateCoordinator(hass, entry)
+
+    await coordinator.async_config_entry_first_refresh()
+
+    entry.runtime_data = coordinator
+    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unload a config entry."""
+    unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+
+    if unload_ok and entry.runtime_data:
+        coordinator = entry.runtime_data
+        await coordinator.async_shutdown()
+        entry.runtime_data = None
```

Comment:
We don't need to reset `runtime_data`. It's deleted by the config entries helper after unload.


```suggestion
```